### DISCUSSION TITLE
Release 1.6.0 & Jakarta

### DISCUSSION THREAD

**Discussion by dkellenb** [mapstruct/mapstruct#3145] [Q&A] [answered]
Hi

Spring 3.x has switched to Jakarta. Currently we cannot migrate to Spring Boot 3.x as long Mapstruct (latest release we use is 1.5.2) is not supporting Jakarta JAXBElement. I've seen that the feature for Jakarata Support (#2730) has already been merged to main. When can we expect Mapstruct 1.6.0 to be released?

Best regards
Daniel

**Comment by filiphr** [comment]
Hey @dkellenb,

You have a valid point. We've been busy with other things and haven't had much time to work on 1.6. I am not exactly sure when we are going to release it, but we are going to talk in the team to do a 1.5.4 release that is going to include #2730.

Cheers,
Filip

**Reply by dkellenb** [reply]
Backporting #2730 to 1.5.x would be great!

**Reply by dkellenb** [reply]
Hi @filiphr . I have seen that 1.5.4 was released, but #2730  seems to not be included into the release. Will it be part of 1.5.5?

**Reply by filiphr** [reply]
Yes I forgot that one. We'll release 1.5.5 with that

**Reply by ghost** [reply, marked as answer]
[1.5.5.Final](https://github.com/mapstruct/mapstruct/releases/tag/1.5.5.Final) is out and includes #2730 🎉

**Reply by dkellenb** [reply]
Good News @JelleDekker-HeadFWD ! We have been able to update our project to Jakarta & SpringBoot 3.0, thanks to release 1.5.5.Final. Good work.

**Reply by oliverdirks** [reply]
Hi @dkellenb,
was it possible to switch to Jakarta & SpringBoot 3.0 with the `@Mapper(componentModel = "spring")` ? Or is it necessary to change form `componentModel = "spring"` to `"jakarta"`
That would lose the possibility of bean injection, wouldn't it?

Cheers 
Olli

**Reply by dkellenb** [reply]
Hi @oliverdirks .
Yes, you can still use `@Mapper(componentModel = "spring")` (or better: `@Mapper(componentModel = ComponentModel.SPRING)`). For spring I also suggest to use `injectionStrategy = InjectionStrategy.CONSTRUCTOR`.
